File: sdk/src/client/bindings/python/iota_client/types/native_token.py
class NativeToken():
    def __init__(self, id, amount):
        """Initialize NativeToken

        Parameters
        ----------
        id : string
            Id of the token
        amount : int
            Native token amount
        """
        self.id = id
        self.amount = amount

    def as_dict(self):
        config = dict(self.__dict__)

        config['amount'] = str(hex(config['amount']))

        return config

File: bindings/nodejs/lib/types/models/native-token.ts
// Copyright 2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { u256 } from '../utils';
import type { HexEncodedString } from '../utils/hex-encoding';
import { Transform } from 'class-transformer';
import { hexToBigInt } from '../utils/hex-encoding';

/**
 * Native token.
 */
export class NativeToken {
    /**
     * Identifier of the native token.
     */
    id!: HexEncodedString;
    /**
     * Amount of native tokens of the given Token ID.
     */
    @Transform((value) => hexToBigInt(value.value))
    amount!: u256;
}
